src/Downloader.ts

```diff
@@ -184,37 +184,56 @@ class Downloader {
     return `${cachedPart}${path}`;
   }
 
-  public async checkCapabilities(): Promise<void> {
-    // check if RESTBase-powered API available
+  public setBaseUrls(baseUrl: string, baseUrlForMainPage: string = this.mw.veApiUrl.href){
+    this.baseUrl = baseUrl;
+    this.baseUrlForMainPage = baseUrlForMainPage;
+  }
+
+  public async checkRestApiAvailabilty(type: string, restApiBase: string, defaultUrlForMainPage: string){
     try {
-      const restApiMainPageQuery = await this.getJSON<any>(`${this.baseUrl}${encodeURIComponent(this.mw.metaData.mainPage)}`);
-      this.mwCapabilities.restApiAvailable = !!restApiMainPageQuery.lead;
+      const restBaseApiQuery = await this.getJSON<any>(`${restApiBase}${type}${encodeURIComponent(this.mw.metaData.mainPage)}`);
+      this.setBaseUrls(`${restApiBase}${type}`);
```

Comment:
Not this should be called outside `checkRestApiAvailabilty` (probably just after).